src/services/MethodService.ts

```diff
@@ -0,0 +1,127 @@
+        this.writer.withIndentationLevel(3, () => this.writer.writeLine(`${parameter?.name ?? `${undefined}`},`));
+        this.writer.withIndentationLevel(3, () => this.writer.writeLine('saveAs'));
+
+        this.writer.writeLine(');');
+    }
+
+    public createMethod(): void {
+        this.writer.writeLine(
+            `return this.${MethodOperation[this.model.operation].toLowerCase()}<${this.getReturnTypeName(
+                this.model.returnType,
+                this.model.returnType?.type.dtoType
+            )}>(`
+        );
+        this.writer.withIndentationLevel(3, () => this.writer.writeLine(`${this.getFullMethodName(this.model)},`));
+        this.model.parameters
+            .filter((z) => z.place === ParameterPlace.Body)
+            .forEach((z) => {
+                this.writer.withIndentationLevel(3, () => this.writer.writeLine(`${z.name},`));
+            });
+
+        if (this.needPipe(this.model.returnType)) {
+            this.writer.writeLine(`).pipe(${this.createPipe(this.model.returnType)});`);
+            return;
+        }
+
+        this.writer.writeLine(');');
+    }
+
+    private needPipe(returnType: IReturnType | undefined): returnType is IReturnType {
+        if (!returnType) {
+            return false;
+        }
+
+        return [PropertyKind.Object, PropertyKind.Identity, PropertyKind.Guid, PropertyKind.Date].includes(returnType.type.kind);
+    }
+
+    private createPipe(returnType: IReturnType): string {
+        if (returnType.type.kind === PropertyKind.Guid) {
+            return `${this.mappersNamespace}.mapGuid()`;
+        }
+
+        if (returnType.type.kind === PropertyKind.Date) {
+            return `${this.mappersNamespace}.mapDate()`;
+        }
+
+        const type = `${this.mappersNamespace}.${returnType.type.type}`;
+        if (returnType.isCollection) {
+            if (returnType.type.kind === PropertyKind.Identity) {
+                return `${this.mappersNamespace}.mapIdentityCollection(${type})`;
+            }
+
+            return `${this.mappersNamespace}.mapCollection(${type})`;
+        }
+
+        if (returnType.type.kind === PropertyKind.Identity) {
+            return `${this.mappersNamespace}.mapIdentitySingle(${type})`;
+        }
+
+        return `${this.mappersNamespace}.mapSingle(${type})`;
+    }
+
+    private getReturnTypeName(returnType: IReturnType | undefined, targetType: string | undefined): string {
+        if (!returnType || !targetType) {
+            return 'void';
+        }
+
+        return this.getFullTypeName(targetType, returnType.isCollection, returnType.isModel);
+    }
+
+    private getFullTypeName(type: string, isCollection: boolean, isModel: boolean): string {
+        const arraySymbol = isCollection ? '[]' : '';
+        return `${isModel ? `${this.modelsNamespace}.` : ''}${type}${arraySymbol}`;
+    }
+
+    public getFullMethodName(model: IMethodModel): string {
```

Comment:
why it public ?

I see you need it only for testing. I guess you should change access to private and you should refactor test

Reply:
I refactor my service and now it's UriBuilder with buildUri public method. I think, it's ok